src/System.Net.Sockets/src/System/Net/Sockets/Socket.cs

```diff
@@ -6083,7 +6141,7 @@ private static bool DoDnsCallback(IAsyncResult result, MultipleAddressConnectAsy
             return DoMultipleAddressConnectCallback(PostOneBeginConnect(context), context);
         }
 
-        private class MultipleAddressConnectAsyncResult : ContextAwareResult
+        private sealed class MultipleAddressConnectAsyncResult : ContextAwareResult
```

Comment:
What is the reason to seal the class?

During work on NetworkInformation I noticed the same changes that had to be reverted.

https://msdn.microsoft.com/en-us/library/ms229023(v=vs.110).aspx
__X DO NOT__ seal classes without having a good reason to do so.